katacomb/katacomb/uv_facade.py

```diff
@@ -401,6 +401,26 @@ def Desc(self):
     def List(self):
         return self.uv.List
 
+    @property
+    def nvis_from_NX(self):
+        """ Derive the number of vis from the attached NX table
+
+            NOTE: Sometimes in certain corner cases the 'nvis'
+            item in self.Desc.Dict incorrectly reports the number
+            of vis in the object (e.g. when the output from UVBlAvg
+            is empty, that happens when all the input data is flagged.).
```

Comment:
I've been grappling with the same question. Lets call it a quirk, not least because my 'bug report' is a private email to BillC. In reality we are sort of abusing the baseline averaging task by relying on it to throw out data that is completely flagged. Its resultant 'empty' UV dataset is not really an expected structure in the AIPS world, thus the need for a workaround.